File: src/dere_plugin/skills/detecting-user-emotions/scripts/get_emotion.py
#!/usr/bin/env python3
"""Query current emotion state from dere daemon."""

from __future__ import annotations

import json
import sys
from pathlib import Path

import requests

# Add parent scripts dir to path
sys.path.insert(0, str(Path(__file__).parent.parent.parent / "scripts"))
from config_reader import get_daemon_url


def get_emotion_state() -> dict | None:
    """Get current emotion state."""
    daemon_url = get_daemon_url()

    try:
        response = requests.get(f"{daemon_url}/emotion/state", timeout=2)
        response.raise_for_status()
        return response.json()
    except requests.exceptions.RequestException as e:
        print(f"Error querying emotion state: {e}", file=sys.stderr)
        return None


def main():
    """Main entry point."""
    state = get_emotion_state()
    if state:
        print(json.dumps(state, indent=2))
    else:
        sys.exit(1)


if __name__ == "__main__":
    main()

File: plugins/dere_core/skills/mood/scripts/get_emotion.ts
import { getDaemonUrl } from "../../../scripts/config_reader.js";

async function getEmotionState(): Promise<Record<string, unknown> | null> {
  const daemonUrl = await getDaemonUrl();
  const controller = new AbortController();
  const timeout = setTimeout(() => controller.abort(), 2_000);
  try {
    const response = await fetch(`${daemonUrl}/emotion/state`, {
      method: "GET",
      signal: controller.signal,
    });
    if (!response.ok) {
      return null;
    }
    return (await response.json()) as Record<string, unknown>;
  } catch {
    return null;
  } finally {
    clearTimeout(timeout);
  }
}

async function main(): Promise<void> {
  const state = await getEmotionState();
  if (state) {
    console.log(JSON.stringify(state, null, 2));
    return;
  }
  process.exit(1);
}

if (import.meta.main) {
  void main();
}
